lib/Benchmark/Executor/Registry.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the PHPBench package
+ *
+ * (c) Daniel Leech <daniel@dantleech.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace PhpBench\Benchmark\Executor;
+
+use PhpBench\Benchmark\ExecutorInterface;
+
+class Registry
```

Comment:
Or we can dispose of the factories all together and created a nested container. One container for the CLI app and another for PHPBench proper.